src/renderer/atlas/shader_ps.hlsl

```diff
@@ -61,49 +60,42 @@ float4 main(float4 pos: SV_Position): SV_Target
 
     // Layer 0:
     // The cell's background color
-    float3 color = decodeRGB(cell.color.y);
+    float4 color = decodeRGBA(cell.color.y);
 
     // Layer 1 (optional):
     // Colored cursors are drawn "in between" the background color and the text of a cell.
     if ((cell.flags & 2) && cursorColor != 0xffffffff)
     {
-        color = lerp(color, decodeRGB(cursorColor), glyphs[cellPos].a);
+        color = alphaBlendPremultiplied(color, decodeRGBA(cursorColor) * glyphs[cellPos].a);
     }
 
     // Layer 2:
     // The cell's glyph potentially drawn in the foreground color
     {
         float4 glyph = glyphs[decodeU16x2(cell.glyphPos) + cellPos];
 
-        if (cell.flags & 1)
+        if ((cell.flags & 1) == 0)
         {
-            // Colored glyphs like Emojis
-            // --> Alpha blending with premultiplied colors in glyph.
-            color = glyph.rgb + color * (1 - glyph.a);
-        }
-        else
-        {
-            // Regular glyphs
-            // --> Alpha blending with regular colors (aka interpolation/lerp).
-            color = lerp(color, decodeRGB(cell.color.x), glyph.a);
+            // Unlike colored glyphs, regular glyphs are mixed with the current foreground color.
+            glyph = decodeRGBA(cell.color.x) * glyph.a;
         }
+
+        color = alphaBlendPremultiplied(color, glyph);
     }
 
     // Layer 3 (optional):
     // Uncolored cursors invert the cells color.
     if ((cell.flags & 2) && cursorColor == 0xffffffff)
     {
-        color = abs(glyphs[cellPos].rgb - color);
+        color.rgb = abs(glyphs[cellPos].rgb - color.rgb);
     }
 
     // Layer 4:
     // The selection is drawn as a semi-transparent, non-premultiplied color on top.
     if (cell.flags & 4)
     {
-        // Selection coloring.
-        float4 sc = decodeRGBA(selectionColor);
-        color = lerp(color, sc.rgb, sc.a);
+        color = alphaBlendPremultiplied(color, decodeRGBA(selectionColor));

```

Comment:
The comment says  "non-premultiplied" but the code says "premultiplied" now!

Reply:
I'll make sure to fix that with my next commit.